File: src/main/java/io/vproxy/vfx/ui/button/TransparentFusionButton.java
package io.vproxy.vfx.ui.button;

import io.vproxy.vfx.theme.Theme;
import javafx.scene.paint.Color;

public class TransparentFusionButton extends FusionButton {
    public TransparentFusionButton() {
        setOnlyAnimateWhenNotClicked(true);
    }

    public TransparentFusionButton(String text) {
        super(text);
        setOnlyAnimateWhenNotClicked(true);
    }

    @Override
    protected Color normalColor() {
        return Theme.current().transparentFusionButtonNormalBackgroundColor();
    }

    @Override
    protected Color hoverColor() {
        return Theme.current().transparentFusionButtonHoverBackgroundColor();
    }

    @Override
    protected Color downColor() {
        return Theme.current().transparentFusionButtonDownBackgroundColor();
    }
}

File: src-ts/src/ui/button/TransparentFusionButton.ts
import { Color } from '../../javafx/color.js';
import { Theme } from '../../theme/Theme.js';
import { FusionButton } from './FusionButton.js';

export class TransparentFusionButton extends FusionButton {
  constructor();
  constructor(text: string);
  constructor(text: string | null = null) {
    super(text);
    this.setOnlyAnimateWhenNotClicked(true);
  }

  protected override normalColor(): Color {
    return Theme.current().transparentFusionButtonNormalBackgroundColor();
  }

  protected override hoverColor(): Color {
    return Theme.current().transparentFusionButtonHoverBackgroundColor();
  }

  protected override downColor(): Color {
    return Theme.current().transparentFusionButtonDownBackgroundColor();
  }
}
